src/test/java/cat/udl/eps/softarch/mypadel/steps/JoinMatchSteps.java

```diff
@@ -405,4 +405,23 @@ public void thereIsAMatchOnAtPmForMinutesAndDeadline(int day, int month, int yea
 
 		matchRepository.save(publicMatch);
 	}
+
+	@Then("^The reservation for this match is cancelled$")
+	public void theReservationForThisMatchIsCancelled() throws Throwable {
+		stepDefs.result = stepDefs.mockMvc.perform(
+			get("/reservations/1")
```

Comment:
You should check it through the relation from Match to Reservation rather than directly from the Reservations collection.